perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonScopedAffiliations.java

```diff
@@ -210,28 +207,133 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 
 	@Override
 	public List<AttributeHandleIdentifier> getHandleIdentifiers() {
+
 		List<AttributeHandleIdentifier> handleIdentifiers = super.getHandleIdentifiers();
+
+		// member related events
 		handleIdentifiers.add(auditEvent -> {
-			if (auditEvent instanceof AllAttributesRemovedForUser) {
-				return ((AllAttributesRemovedForUser) auditEvent).getUser().getId();
-			} else {
-				return null;
+
+			if (auditEvent instanceof DirectMemberAddedToGroup) {
+				return ((DirectMemberAddedToGroup) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof IndirectMemberAddedToGroup) {
+				return ((IndirectMemberAddedToGroup) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberRemovedFromGroupTotally) {
+				return ((MemberRemovedFromGroupTotally) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberExpiredInGroup) {
+				return ((MemberExpiredInGroup) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberValidatedInGroup) {
+				return ((MemberValidatedInGroup) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberValidated) {
+				return ((MemberValidated) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberExpired) {
+				return ((MemberExpired) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberSuspended) {
+				return ((MemberSuspended) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberDisabled) {
+				return ((MemberDisabled) auditEvent).getMember().getUserId();
+			} else if (auditEvent instanceof MemberInvalidated) {
+				return ((MemberInvalidated) auditEvent).getMember().getUserId();
 			}
+
+			// no match
+			return null;
+
 		});
-		handleIdentifiers.add(auditEvent -> {
-			if (auditEvent instanceof AttributeSetForUser && ((AttributeSetForUser) auditEvent).getAttribute().getFriendlyName().equals(getSecondarySourceAttributeFriendlyName())) {
-				return ((AttributeSetForUser) auditEvent).getUser().getId();
-			} else {
-				return null;
+
+		return handleIdentifiers;
+	}
```

Comment:
I am not sure if this will work when you are overriding method resolveVirtualAttributeValueChange.

Reply:
I believe, calling ```super.resolveVirtualAttributeValueChange()``` at the start of own implementation of ``resolveVirtualAttributeValueChange()`` should still in its logic call ```getHandleIdentifiers()``` on actual module/class, since its overridden too. Inside of it it calls super to add generic message handlers, then adds our member messages handlers and everything is processed by super logic of resolveVirtualAttributeValueChange. Then it process group messages and returns resolved messages.

Reply:
My bad, I missed that line.